autoscale-docker-swarm-container/README.md

```diff
@@ -0,0 +1,167 @@
+container that includes the CAF autoscale application along with
+standard modules required to scale a service based on rabbitmq metrics.
+
+To get the autoscale application up and running look at the [Quick Start](#quick-start) section below.
+
+
+### Container deployment
+
+### Configuration
+
+Configuration of the AutoScaler is supported through the following environment variables:
+
+ - `DOCKER_HOST`  
+    Default: `unix:///var/run/docker.sock`  
+    Used to specify the Docker Swarm REST endpoint.  Supports unix sockets and tcp type connections e.g. http://machine:2375.
+
+ - `CAF_RABBITMQ_MGMT_URL`  
+    Default: `http://rabbitmq:15672`  
+    Used to specify the RabbitMQ Management API Endpoint.  Alternatively `CAF_RABBITMQ_HOST` and `CAF_RABBITMQ_MGMT_PORT` may instead be specified individually.
+
+ - `CAF_RABBITMQ_MGMT_USERNAME`  
+    Default: `guest`  
+    Used to specify the username used to connect to RabbitMQ.  If `CAF_RABBITMQ_MGMT_USERNAME` is not specified then `CAF_RABBITMQ_USERNAME` will also be checked before falling back to the default.
+
+ - `CAF_RABBITMQ_MGMT_PASSWORD`  
+    Default: `guest`  
+    Used to specify the password used to connect to RabbitMQ.  If `CAF_RABBITMQ_MGMT_PASSWORD` is not specified then `CAF_RABBITMQ_PASSWORD` will also be checked before falling back to the default.
+
+ - `CAF_AUTOSCALER_DOCKER_SWARM_STACK`  
+    Required property used to specify the docker stack(s) which the autoscaler is to monitor.  Only applications deployed directly or indirectly within these stacks are auto-scaled.
+    To specify multiple stacks use a comma delimted list e.g. "JobService,DataProcessing".
+
+ - `CAF_AUTOSCALER_SCALING_DELAY`  
+    Minimum: `1`  
+    Default: `10`  
+    Used to specify the number of queue statistics records that must be acquired sequentially before performing analysis upon them and issuing a scaling recommendation. This means the period in time between making scaling recommendations is the service interval multiplied by the scalingDelay. This number should be small enough to respond to varying load but big enough to avoid erratic behaviour.
+
+ - `CAF_AUTOSCALER_BACKLOG_GOAL`  
+    Default: `300`  
+    Used to specify the amount of time, in seconds, that we ideally want to complete the current backlog of messages in. This is effectively a quality of service parameter, where lower will trigger more aggressive scaling.
+
+ - `CAF_AUTOSCALER_MAXIMUM_INSTANCES`  
+    Default: `100`  
+    Used to specify the maximum number of instances that any worker can be scaled to.
+
+ - `CAF_DOCKER_SWARM_TIMEOUT`  
+    Default: `30`  
+    Used to specify the max length of time in seconds that a docker REST call can take before a timeout out occurs.  
+
+ -  `CAF_DOCKER_SWARM_HEALTHCHECK_TIMEOUT`  
+    Default: `5`  
+    Used to specify the max length of time in seconds that the Docker endpoint healthcheck can take before a timeout occurs.
```

Comment:
Healthchecks exist in both the source and scaler modules for marathon, so I have duplicated the same healthchecks to the swarm node.
https://github.com/Autoscaler/autoscaler/blob/develop/autoscale-scaler-marathon/src/main/java/com/hpe/caf/autoscale/scaler/marathon/MarathonServiceScaler.java#L125